hypothesis-python/src/hypothesis/database.py

```diff
@@ -146,10 +146,47 @@ def __call__(self, *args: Any, **kwargs: Any) -> "ExampleDatabase":
 
 
 class ExampleDatabase(metaclass=_EDMeta):
-    """An abstract base class for storing examples in Hypothesis' internal format.
+    """
+    A Hypothesis database, for use in the |settings.database| setting.
+
+    A Hypothesis database acts like a ``Mapping[bytes, set[bytes]]``, mapping
+    each ``bytes`` key to a set of ``bytes`` values.
+
+    Change listening
+    ----------------
+
+    An optional extension to |ExampleDatabase| is change listening. On databases
+    which support it, you can call |ExampleDatabase.add_listener| to add a function
+    as a change listener, which will be called whenever a value is added, deleted,
+    or moved. See |ExampleDatabase.add_listener| for details.
+
+    All databases in Hypothesis support change listening. Custom database classes
+    are not required to sipport change listening, unless they want to be
+    compatible with features that require change listening. Currently, no Hypothesis
+    features require change listening.
+
+    .. note::
+
+        Change listening is required by `HypoFuzz <https://hypofuzz.com/>`_.
+
+    Methods
+    -------
+
+    Required methods:
+
+    * |ExampleDatabase.save|
+    * |ExampleDatabase.fetch|
+    * |ExampleDatabase.delete|
+
+    Optional methods:
+
+    * |ExampleDatabase.move|
 
-    An ExampleDatabase maps each ``bytes`` key to many distinct ``bytes``
-    values, like a ``Mapping[bytes, set[bytes]]``.
+    Change listening methods:
+
+    * |ExampleDatabase.add_listener|
+    * |ExampleDatabase.remove_listener|
+    * |ExampleDatabase.clear_listeners|
```

Comment:
- I'd just list these under "Optional methods"
- shouldn't we be listing `._start_listening()` and `._stop_listening()` as the things-to-implement here, and autodoc-ing them?  (or direct calls to `._broadcast_change()`, if it's cheap)  The add/remove/clear methods can just be inherited, I think.